### PR TITLE
[NEP-17596] Superset: Export Dashboards should save zip file and clean out prev version totally

### PR DESCRIPTION
**Pull request by barrynguyen8**
Ticket: https://jobready.atlassian.net/browse/NEP-17596

### CODE REVIEW

File: lib/superset/dashboard/export.rb [hunk 1]
```diff
@@ -45,35 +65,74 @@ def params
...
+          relative_path = existing_path.relative_path_from(destination).to_s.downcase
+          puts "Checking file: #{relative_path}"
+
+          unless latest_files_set.include?(relative_path)
+            puts "Removing file: #{existing_file} as it's not in the latest export."
+            FileUtils.rm_f(existing_file)
+          else
+            puts "Retaining file: #{existing_file} as it's part of the latest export."
+          end
+        end
```
Comment (jbat) [inline lib/superset/dashboard/export.rb:127]: @barrynguyen8  there is a fair chunk of logic here. 
I had an idea .....

Q: Given we always copy over the entire new extracted file ... 
do we care about any thing that is in the destination path prior to the copy?

ie ... we always copy over the current state of a dashboard, where all files are considered necessary.
So why not just delete the entire contents of the destination path before we copy across?
No checks / comparisons needed.

```
# something like
Dir.glob("#{destination_path_with_dash_id}/*").each do |file|
  FileUtils.rm(file) if File.file?(file)
end

copy_export_files_to_destination_path
```

File: lib/superset/dashboard/export.rb [hunk 2]
```diff
@@ -23,10 +23,26 @@ def initialize(dashboard_id: , destination_path: )
...
+        puts "Temporary directory created at: #{tmp_uniq_dashboard_path}"
+
         save_exported_zip_file
+        puts "Exported zip file saved at: #{zip_file_name}"
+
         unzip_files
-        copy_export_files_to_destination_path if destination_path
+        puts "Files unzipped to: #{tmp_uniq_dashboard_path}"
+
+        copy_export_files_to_destination_path
```
Comment (jbat) [inline lib/superset/dashboard/export.rb:36]: @barrynguyen8  ... any particular reason why you removed the condition here?

```
copy_export_files_to_destination_path if destination_path
```

Comment (barrynguyen8) [reply lib/superset/dashboard/export.rb:36]: Accident, sorry!

Comment (barrynguyen8) [reply lib/superset/dashboard/export.rb]: Reverted https://github.com/rdytech/superset-client/pull/40/commits/8a16a72c90af6b9fccc743505864b77c9021b8fa

File: lib/superset/dashboard/export.rb (hunk 1 above)
Comment (barrynguyen8) [reply lib/superset/dashboard/export.rb:118]: @jbat how does this new approach seem? https://github.com/rdytech/superset-client/pull/40/commits/9b89c0b4e69f794cbf6b483efeee1b31ee9cf755

File: lib/superset/dashboard/export.rb [hunk 3]
```diff
@@ -45,35 +72,58 @@ def params
...
+          next if File.directory?(item) # Skip directories
+
+          relative_item_path = Pathname.new(item).relative_path_from(Pathname.new(download_folder)).to_s
+          destination_item = File.join(destination_path_with_dash_id, relative_item_path)
 
-        Dir.glob("#{download_folder}/*").each do |item|
-          FileUtils.cp_r(item, path_with_dash_id)
+          FileUtils.mkdir_p(File.dirname(destination_item)) unless Dir.exist?(File.dirname(destination_item))
+          puts "Copying #{item} to #{destination_item}"
+          FileUtils.cp(item, destination_item)
```
Comment (jbat) [inline lib/superset/dashboard/export.rb:110]: best to revert this to the original cp_r 
as its cleaner

File: lib/superset/dashboard/export.rb [hunk 4]
```diff
@@ -23,12 +23,35 @@ def initialize(dashboard_id: , destination_path: )
...
+        puts "Export process completed successfully."
+
+        # Return the list of copied files
+        Dir.glob("#{destination_path_with_dash_id}/**/*").select { |f| File.file?(f) }
+      rescue StandardError => e
+        puts "Export failed: #{e.message}"
+        raise
+      ensure
+        cleanup_temp_dir
       end
```
Comment (jbat) [inline lib/superset/dashboard/export.rb:51]: @barrynguyen8 
please add a logger and move all these puts .. in a logger msg 
similar to this
https://github.com/rdytech/superset-client/blob/develop/lib/superset/dataset/warm_up_cache.rb#L36C5-L38C10

File: lib/superset/dashboard/export.rb (hunk 3 above)
Comment (barrynguyen8) [reply lib/superset/dashboard/export.rb]: Reverted: https://github.com/rdytech/superset-client/pull/40/commits/6e45818648453c83347db7ac5ea5b03076b5d15f

File: lib/superset/dashboard/export.rb (hunk 4 above)
Comment (barrynguyen8) [reply lib/superset/dashboard/export.rb:38]: @jbat added https://github.com/rdytech/superset-client/pull/40/commits/0dc30db5d997209a4a3167aa9735aa5963278c3b

**Review by jbat — APPROVED** [review]

**Pull request merged**